edith-step/src/components/TopNavbar.test.js

```diff
@@ -1,39 +1,112 @@
-});
+describe('When not logged in, Top Navigation Bar', () => {
+  beforeEach(() => {
+    enableFetchMocks();
+    component = mount(<TopNavbar />);
+    component.setState({user: null});
+  });
 
-// Checks if the Navigation Bar is rendered.
-test('renders', () => {
-  expect(component.exists()).toBe(true);
-});
+  afterEach(() => {
+    component.unmount();
+  });
 
-// Checks if the navigation Bar provides three different options.
-test('has three options', () => {
-  expect(component.find('.links').length).toBe(3);
-});
+  // Checks if the Navigation Bar is rendered.
+  test('renders', () => {
+    expect(component.exists()).toBe(true);
+  });
 
-// Checks if there is Home option.
-test('has Home option', () => {
-  expect(component.find('.home').exists()).toBe(true);
-});
+  // Checks if the navigation Bar provides three different options.
+  test('has three options', () => {
+    expect(component.find('.links').length).toBe(3);
+  });
+
+  // Checks if there is Home option.
+  test('has Home option', () => {
+    expect(component.find('.home').exists()).toBe(true);
+  });
+
+  // Checks if there is Features option.
+  test('has Features option', () => {
+    expect(component.find('.features').exists()).toBe(true);
+  });
+
+  // Checks if there is Login option.
+  test('has Login option', () => {
+    expect(component.find('.login-button').exists()).toBe(true);
+  });
 
-// Checks if there is Features option.
-test('has Features option', () => {
-  expect(component.find('.features').exists()).toBe(true);
+  // Checks if there is Dashboard option.
+  test('has Dashboard section', () => {
+    expect(component.find('.dashboard').exists()).toBe(false);
+  });
+
+  // Checks if dropdown is shown.
+  test('should not display dropdown', () => {
+    expect(component.find('.dropdowns').exists()).toBe(false);
+  });
 });
 
-// Checks if there is Login option.
-test('has Login option', () => {
-  expect(component.find('.login-button').exists()).toBe(true);
+describe('When logged in, Top Navigation Bar', () => {
+  beforeEach(() => {
+    enableFetchMocks();
+    component = mount(<TopNavbar />);
+    handleModalClose = jest.spyOn(TopNavbar.prototype, 'handleModalClose');
+    component.setState({user: userObj});
+  });
+
+  afterEach(() => {
+    component.unmount();
+  });
+
+  // Checks if the Navigation Bar is rendered.
+  test('renders', () => {
+    expect(component.exists()).toBe(true);
+  });
+
+  // Checks if there is Dashboard option.
+  test('has Dashboard section', () => {
+    expect(component.find('.dashboard').exists()).toBe(true);
+  });
+
+  // Checks if there is Login option.
+  test('does not have Login option', () => {
+    expect(component.find('.login-button').exists()).toBe(false);
+  });
+
+  // Checks if there is Dropdown menu.
+  test('has Dropdown menu', () => {
+    expect(component.find('.dropdowns').exists()).toBe(true);
+  });
+
+  test('opens upload modal box when upload receipt button is clicked', () => {
```

Comment:
Nit: You could add a comment for this and the next test just to keep it consistent with the rest of the file.

Reply:
Most Definitely.